acir/src/native_types/witness.rs

```diff
@@ -27,23 +27,75 @@ impl Witness {
     pub const fn can_defer_constraint(&self) -> bool {
         true
     }
+}
+
+impl From<u32> for Witness {
+    fn from(value: u32) -> Self {
+        Self(value)
+    }
+}
+
+/// A map from the witnesses in a constraint system to the field element values
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, Default, Serialize, Deserialize)]
+pub struct WitnessMap(BTreeMap<Witness, acir_field::FieldElement>);
+
+impl WitnessMap {
+    pub fn new() -> Self {
+        Self(BTreeMap::new())
+    }
+    pub fn get(&self, witness: &Witness) -> Option<&acir_field::FieldElement> {
+        self.0.get(witness)
+    }
+    pub fn get_index(&self, index: u32) -> Option<&acir_field::FieldElement> {
+        self.0.get(&index.into())
+    }
+    pub fn contains_key(&self, key: &Witness) -> bool {
+        self.0.contains_key(key)
+    }
+    pub fn insert(
+        &mut self,
+        key: Witness,
+        value: acir_field::FieldElement,
+    ) -> Option<acir_field::FieldElement> {
+        self.0.insert(key, value)
+    }
+}
 
-    pub fn to_bytes(
-        witnesses: &std::collections::BTreeMap<Witness, acir_field::FieldElement>,
-    ) -> Vec<u8> {
-        let buf = rmp_serde::to_vec(witnesses).unwrap();
+impl Index<&Witness> for WitnessMap {
+    type Output = acir_field::FieldElement;
+
+    fn index(&self, index: &Witness) -> &Self::Output {
+        &self.0[index]
+    }
+}
+
+impl From<BTreeMap<Witness, acir_field::FieldElement>> for WitnessMap {
+    fn from(value: BTreeMap<Witness, acir_field::FieldElement>) -> Self {
+        Self(value)
+    }
+}
+
+impl From<WitnessMap> for Vec<u8> {
```

Comment:
These should implement `TryFrom` rather than `From`

Reply:
Yes, I built this on top of 0.10.3 instead of the pre-0.11 so we don't have fallible functions everywhere so I figured unwrap is easiest. Probably need to finish and land https://github.com/noir-lang/aztec_backend/pull/151 so I can avoid unwrapping.

Reply:
Done and rebased on master